test/vfd.c

```diff
@@ -3915,33 +4032,1949 @@ test_ctl(void)
+ * Function:    test_vector_io__write_v_indiv
+ *
+ * Purpose:     Write the supplied vector as a sequence of individual
+ *              writes.
+ *
+ *              All arrays parameters are presumed to be of length
+ *              count.
+ *
+ * Return:      Return TRUE if successful, and FALSE if any errors
+ *              are encountered.
+ *
+ * Programmer:  John Mainzer
+ *              6/21/20
+ *
+ * Modifications:
+ *
+ *              None.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+static hbool_t
+test_vector_io__write_v_indiv(H5FD_t *lf, uint32_t count, H5FD_mem_t types[], haddr_t addrs[], size_t sizes[],
+                              void *write_bufs[])
+{
+    hbool_t    size_fixed = FALSE;
+    hbool_t    type_fixed = FALSE;
+    hbool_t    result     = TRUE; /* will set to FALSE on failure */
+    hbool_t    verbose    = FALSE;
+    uint32_t   i;
+    size_t     size = ULONG_MAX;
+    H5FD_mem_t type = H5FD_MEM_NTYPES;
+
+    for (i = 0; i < count; i++) {
+
+        SET_SIZE(size_fixed, sizes, size, i);
+
+        SET_TYPE(type_fixed, types, type, i);
+
+        if (H5FDwrite(lf, type, H5P_DEFAULT, addrs[i], size, write_bufs[i]) < 0) {
+
+            if (verbose) {
+
+                HDfprintf(stdout, "%s: HDwrite() failed on entry %d.\n", __func__, i);
+            }
+            result = FALSE;
+            break;
+        }
+    }
+
+    return (result);
+
+} /* end test_vector_io__write_v_indiv() */
+
+/*-------------------------------------------------------------------------
+ *
+ * Function:    test_vector_io__verify_v
+ *
+ * Purpose:     Verify that the read and write buffers of the supplied
+ *              vectors are identical.
+ *
+ * Return:      TRUE if the read and write vectors are identical, and
+ *              FALSE otherwise.
+ *
+ * Programmer:  John Mainzer
+ *              6/21/20
+ *
+ * Changes:     None.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+static hbool_t
+test_vector_io__verify_v(uint32_t count, H5FD_mem_t types[], size_t sizes[], void *write_bufs[],
+                         void *read_bufs[], const char *name)
+{
+    hbool_t     size_fixed = FALSE;
+    hbool_t     type_fixed = FALSE;
+    hbool_t     identical  = TRUE;
+    hbool_t     verbose    = TRUE;
+    uint32_t    i;
+    uint32_t    j;
+    uint32_t    buf_size;
+    char *      w_buf;
+    char *      r_buf;
+    const char *mem_type_names[7] = {"H5FD_MEM_DEFAULT", "H5FD_MEM_SUPER", "H5FD_MEM_BTREE", "H5FD_MEM_DRAW",
+                                     "H5FD_MEM_GHEAP",   "H5FD_MEM_LHEAP", "H5FD_MEM_OHDR"};
+    size_t      size              = ULONG_MAX;
+    H5FD_mem_t  type              = H5FD_MEM_NTYPES;
+
+    i = 0;
+
+    while ((i < count) && (identical)) {
+
+        SET_SIZE(size_fixed, sizes, size, i);
+
+        SET_TYPE(type_fixed, types, type, i);
+
+        buf_size = (uint32_t)(size);
```

Comment:
What is the purpose of this? Why not just use size?

Reply:
Looks like the printf format specifiers don't quite match the parameter types, I'll fix that. I recall there was some disagreement about whether we can use %z or %j, so I guess I'll avoid those for now.